$NetBSD: patch-Makefile,v 1.1 2018/10/13 17:02:17 nia Exp $

Don't override pkgsrc-set variables.

Don't make assumptions based on uname.

--- Makefile.orig	2018-10-12 17:39:34.000000000 +0000
+++ Makefile
@@ -20,19 +20,6 @@ HAVE_CHD      := 1
 
 TARGET_NAME   := reicast
 
-CXX      = ${CC_PREFIX}g++
-CC       = ${CC_PREFIX}gcc
-CC_AS    = ${CC_PREFIX}as
-
-MFLAGS   := 
-ASFLAGS  := 
-LDFLAGS  :=
-LDFLAGS_END :=
-INCFLAGS :=
-LIBS     :=
-CFLAGS   := 
-CXXFLAGS :=
-
 GIT_VERSION := " $(shell git rev-parse --short HEAD || echo unknown)"
 ifneq ($(GIT_VERSION)," unknown")
 	CXXFLAGS += -DGIT_VERSION=\"$(GIT_VERSION)\"
@@ -42,15 +29,6 @@ UNAME=$(shell uname -a)
 
 LIBRETRO_DIR := .
 
-# Cross compile ?
-
-ifeq (,$(ARCH))
-	ARCH = $(shell uname -m)
-endif
-
-# Target Dynarec
-WITH_DYNAREC = $(ARCH)
-
 ifeq ($(ARCH), $(filter $(ARCH), i386 i686))
 	WITH_DYNAREC = x86
 endif
@@ -635,7 +613,7 @@ RZDCY_CXXFLAGS += $(CORE_DEFINES)
 CFLAGS         += $(CORE_DEFINES)
 CXXFLAGS       += $(CORE_DEFINES)
 
-CFLAGS   += $(OPTFLAGS) -c
+CFLAGS   += -c
 CFLAGS   += -fno-strict-aliasing -ffast-math
 CXXFLAGS += -fno-rtti -fpermissive -fno-operator-names
 LIBS     += -lm 
@@ -682,14 +660,14 @@ $(TARGET): $(OBJECTS)
 ifeq ($(STATIC_LINKING), 1)
 	$(AR) rcs $@ $(OBJECTS)
 else
-	$(LD) $(MFLAGS) $(fpic) $(SHARED) $(LDFLAGS) $(OBJECTS) $(LDFLAGS_END) $(GL_LIB) $(LIBS) -o $@
+	$(LD) $(fpic) $(SHARED) $(LDFLAGS) $(OBJECTS) $(LDFLAGS_END) $(GL_LIB) $(LIBS) -o $@
 endif
 
 %.o: %.cpp
-	$(CXX) $(INCFLAGS) $(CFLAGS) $(MFLAGS) $(CXXFLAGS) $< -o $@
+	$(CXX) $(INCFLAGS) $(CFLAGS) $(CXXFLAGS) $< -o $@
 	
 %.o: %.c
-	$(CC) $(INCFLAGS) $(CFLAGS) $(MFLAGS) $< -o $@
+	$(CC) $(INCFLAGS) $(CFLAGS) $< -o $@
 
 %.o: %.S
 	$(CC_AS) $(ASFLAGS) $(INCFLAGS) $< -o $@
